sources/mongodb/helpers.py

```diff
@@ -0,0 +1,115 @@
+"""Mongo database source helpers"""
+
+from itertools import islice
+from typing import Any, Dict, Iterator, Optional
+
+import dlt
+from bson.decimal128 import Decimal128
+from bson.objectid import ObjectId
+from dlt.common.configuration.specs import BaseConfiguration, configspec
+from dlt.common.typing import TDataItem
+from pymongo import MongoClient
+
+
+CHUNK_SIZE = 10000
+
```

Comment:
let's try to type client, collection and cursor anyway by doing that only when typechecking

from typing import TYPE_CHECKING
if TYPE_CHECKING:
    TMongoClient = MongoClient[Any]
    TCollection = Collection[Any]  # type: ignore
    TCursor = Cursor[Any]
else:
    TMongoClient = Any
    TCollection = Any
    TCursor = Any

and use TMongoClient, TCollection and TCursor below. this should work!